libnmstate/state.py

```diff
@@ -144,6 +185,24 @@ def verify_interfaces(self, other_state):
 
         self._assert_interfaces_equal(other_state)
 
+    def verify_routes(self, other_state):
+        """
+        Verify that the self state and the other_state are identical.
+        """
+        for iface_name, routes in six.viewitems(self.iface_routes):
+            routes.sort(key=_route_sort_key)
```

Comment:
This should have been done when indexing the routes (and we will not need to do this to `other` as well).

Reply:
The `metadata.py` `_generate_route_metadata()` use the `iface_routes` which does not require ordering.
The only ordering is required just for verification.